landbosse/excelio/XlsxManagerRunner.py

```diff
@@ -107,3 +117,70 @@ def extract_details_lists(self, runs_dict):
                 if key.endswith('_csv'):
                     runs_for_csv.extend(value)
         return runs_for_csv
+
+    def read_project_and_parametric_list_from_xlsx(self):
+        """
+        This method reads both the project and parametric list from the
+        project_list xlsx. It returns them as a tuple.
+
+        To ensure backward compatibility with project list spreadsheets
+        that contain only one sheet, the following actions take place
+
+        1. Project list spreadsheet contains one sheet: The only sheet
+        that is present is read as the project list, and a dataframe for
+        parametrics is returned that has all the columns but no rows so
+        it can be used in joins. It doe snot matter what the sheet name
+        is.
+
+        2. Project list spreadsheet contains two sheets: The sheet named
+        "Project list" is assumed to be the project list and the sheet
+        named "Parametric list" is assumed to be the specifications for
+        the parametric variables. If one or both of those tab names are
+        missing, an exception is raised
+
+        Returns
+        -------
+        pandas.DataFrame
+            The enhanced project list that has support for all parametric
+            adjustments for each step.
+
+        Raises
+        ------
+        KeyError
+            When the spreadsheet contains multiple sheets and one or
+            both of "Project list" or "Parametric list" are undefined.
+        """
+        path_to_project_list = self.file_ops.landbosse_input_dir()
+        sheets = XlsxDataframeCache.read_all_sheets_from_xlsx('project_list', path_to_project_list)
+
+        # If there is one sheet
+        if len(sheets.values()) == 1:
+            first_sheet = list(sheets.values())[0]
+            project_list = first_sheet
+            parametric_list = pd.DataFrame(columns=[
+                'Project ID',
+                'Dataframe sheet name',
+                'Row name',
+                'Col name',
+                'Start',
+                'End'
+            ])
+
+        # If the parametric and project lists exist, read them
+        elif 'Parametric list' in sheets.keys() and 'Project list' in sheets.keys():
+            project_list = sheets['Project list']
+            parametric_list = sheets['Parametric list']
+
+        # Otherwise, raise an exception
+        else:
+            raise KeyError("Project list needs to have a single sheet or sheets named 'Project list' and 'Parametric list'.")
+
+        # Instantiate and XlsxReader to assemble master input dictionary
+        xlsx_reader = XlsxReader()
+
+        # Join in the parametric variable modifications
+        parametric_value_list = xlsx_reader.create_parametric_value_list(parametric_list)
+        enhanced_project_list = xlsx_reader.outer_join_projects_to_parametric_values(project_list,
+                                                                                 parametric_value_list)
```

Comment:
It looks like the parametric_value_list is empty when it gets called here using the old data so it's probably a problem with the xlsx_reader.create_parametric_value_list function. Would you review and revise? I can try to fix it myself as well but I don't have time this evening so I thought I'd see if you happen to have time today. No worries if not, I'll resume my review on Monday either way and can make revisions then if needed.